x-pack/plugins/infra/public/alerting/metric_threshold/components/expression.tsx

```diff
@@ -255,17 +222,26 @@ export const Expressions: React.FC<Props> = props => {
               setAlertParams={updateParams}
               errors={errors[idx] || emptyError}
               expression={e || {}}
-            />
+            >
+              <ExpressionChart
+                loading={chartData.loading}
+                data={chartData.data}
+                id={idx}
+                expression={e}
+              />
```

Comment:
The chart flashes with `Oops! No data...` before it switches to loading. For some reason `chartData.loading` seems to be `false` when the flyout initially loads.